cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/WorkerTwillProgramController.java

```diff
@@ -45,11 +44,11 @@ protected void doCommand(String name, Object value) throws Exception {
       return;
     }
 
-    Map<String, Integer> command = (Map<String, Integer>) value;
+    Map<String, String> command = (Map<String, String>) value;
     try {
-      for (Map.Entry<String, Integer> entry : command.entrySet()) {
+      for (Map.Entry<String, String> entry : command.entrySet()) {
         LOG.info("Changing worker instance count: {} new count is: {}", entry.getKey(), entry.getValue());
-        changeInstances(entry.getKey(), entry.getValue());
+        changeInstances(entry.getKey(), Integer.valueOf(entry.getValue()));
```

Comment:
We always verify the string is derived from an integer before passing it on to the controller. We can change it to always pass in integer.